b2b_ecommerce/views.py

```diff
@@ -119,6 +120,21 @@ def get(self, request, *args, **kwargs):
         order_hash = kwargs["hash"]
         order = get_object_or_404(B2BOrder, unique_id=order_hash)
 
+        receipt = B2BReceipt.objects.filter(order=order).order_by("-created_on").first()
+        receipt_data = {"card_number": None, "card_type": None}
+        if receipt and (
+            "req_card_number" in receipt.data or "req_card_type" in receipt.data
```

Comment:
This is ok as per functionality but this line can be removed because we are checking it twice, once here and once inside.